openupgradelib/openupgrade.py

```diff
@@ -834,6 +837,48 @@ def reactivate_workflow_transitions(cr, transition_conditions):
             (condition, transition_id))
 
 
+def log():
+    """Decorator for automatic logging of the executed method.
+    :param custom_logger: for defining an specific logger
+    :param details: if True, more details are given in the log"""
+    def wrap(func, custom_logger=None, details=False):
+        def wrapped_function(*args, **kwargs):
+            msg = "Executing method %s" % func.__name__
+            if details:
+                if args:
+                    msg += " with args %s" % str(args)
+                if kwargs:
+                    if args:
+                        msg += " and kwargs %s" % str(kwargs)
+                    else:
+                        msg += " with kwargs %s" % str(kwargs)
+            custom_logger.info(msg)
+            func(*args, **kwargs)
+        if custom_logger is None:
+            custom_logger = logger
+        return wrapped_function
+    return wrap
+
+
+def logged_progress(func, iterator, title="", args=None, kwargs=None):
+    """Log the progress of a method that it's run across the elements of an
+    iterator. The passed method should have the individual element to treat
+    as the first argument. It can have more arguments that are passed as
+    a list and a dictionary arguments for the args and kwargs respectively.
+    This is due to the extra title argument, that doesn't allow the
+    "addressing" of the arguments.
+    """
+    widgets = [title + ' ', progressbar.Percentage(), ' ', progressbar.Bar(),
+               ' ', progressbar.ETA()]
```

Comment:
Consider stripping the title of white space at least on the right side and also consider the case where title is blank where we would not want a space. 

Also, why is the blank space appended to title, when it could just be it's own element of the list. There could be cases where title has a string representation but cannot be appended to a str.